UI - new Icon and Chevron components

meirish opened the pull request:
This is a follow-up to https://github.com/hashicorp/vault/pull/6629. 

This PR implements a new `Icon` component and a variant of the `Icon` component: `Chevron`. The new icons use [`ember-svg-jar`](https://github.com/ivanvotti/ember-svg-jar) to inline SVG files from a set of configured paths. This add-on also gives us a handy ember app that lets us browse all of the discovered SVGs while the development server is running - to see it, run the ember dev server as usual (`yarn start` from the `ui` dir) and then navigate to http://localhost:4200/ui/ember-svg-jar/index.html . The new Icon also sources most of the SVGs from HashiCorp's design system, Structure - https://www.npmjs.com/package/@hashicorp/structure-icons . 

These replace the usage of the old `i-con` / `ICon` component. The old component's implementation relied on using partials to inline the SVGs, which will be deprecated in a future ember release and don't play well with ember-engines. Any SVGs used, but not provided by Structure were renamed to .svg and moved to the `public/` folder. The images used for the various types of mounts were moved to `public/eco/` - they were previously in `/templates/svg/enable/`. The rest of the partials in `/templates/svg/` were removed.

TODO - this before merging this branch 
- [x] Remove old ICon component 
- [x] Add new components to Storybook
- [x] Add tests for new components
- [x] 💃 🕺 👯‍♂ 👯 

Following this, I think it makes sense to get this feature branch merged to master, and make progress on the auth/engines settings engine after that (that is the last check item on #6629 's TODO list).

noelledaley commented on ui/app/styles/components/env-banner.scss:
```diff
@@ -1,10 +1,24 @@
+  align-self: center;
+  border-radius: 3rem;
+  background: linear-gradient(135deg, $blue, $purple);
+  animation: env-banner-color-rotate 8s infinite linear alternate;
```
aww yesss, very excited about this 🌈

noelledaley commented on ui/lib/core/addon/templates/components/icon.hbs:
```diff
@@ -0,0 +1,3 @@
+<span class="hs-icon {{class}}" ...attributes>
```
what does `hs` stand for? hashicorp structure? 😛

meirish replied:
Yep!

noelledaley replied:
`vlt-` is a little more clear to me, but 🤷‍♀

meirish commented on ui/lib/core/addon/components/chevron.js:
```diff
@@ -0,0 +1,34 @@
+  isButton: false,
+  glyph: computed('direction', function() {
+    let { direction } = this;
+    assert(
```
Yeah it's a lot like `invariant` in React land. We don't use it a ton, but they get stripped in prod builds so it's handy. There's some other levels too: https://api.emberjs.com/ember/release/modules/@ember%2Fdebug

meirish commented on ui/app/templates/components/console/log-error-with-html.hbs:
```diff
@@ -1,5 +1,5 @@
 <div class="console-ui-alert has-text-danger">
-  {{i-con glyph="close-circled" aria-hidden="true" size=12}}
+  <Icon @glyph="cancel-circle-fill" aria-hidden="true" />
   <pre>{{{content}}}</pre>
```
Yeah, since this is possibly inserting HTML, we don't want to escape at the template layer. I think the only time this is really used is in the control group flow - if you hit a Control Group, we log the token and a link to the lookup page right in the console.

madalynrose commented on ui/tests/integration/components/icon-test.js:
```diff
@@ -0,0 +1,26 @@
+    // Set any properties with this.set('myProperty', 'value');
+    // Handle any actions with this.set('myAction', function(val) { ... });
+
+    await render(hbs`{{ic-on}}`);
```
where is `ic-on` coming from? I haven't seen it called this anywhere.

meirish replied:
Whoops, ember generators still don't like single-word components, so this was the generated test for icon.js - I'll update all of this before finalizing the PR!

madalynrose commented on ui/tests/integration/components/icon-test.js:
```diff
@@ -0,0 +1,26 @@
+
+    await render(hbs`{{ic-on}}`);
+
+    assert.equal(this.element.textContent.trim(), '');
```
should we also be asserting that the correct classes are applied and aria-hidden is set correctly?

madalynrose commented:
This looks great! Super excited about it. bonus points for all those deletions!

meirish commented:
OK, sorry for the rebase / force-push, I know that can sometimes lead to weirdness with tracking comments, etc. I rebased the parent branch first, and had to move a few things around to get tests back to 🍏 - that PR is here: https://github.com/hashicorp/vault/pull/6712. 

After that, I rebased this PR on `ui-ember-engines` and force-pushed, so this PR is now up to date with master and includes the toolbar changes ✨ .

meirish commented:
Gonna merge and then PR to master! 🎉
